src/components/TransactionRow.tsx

```diff
@@ -6,22 +6,40 @@ import {
   type ParseEventLogsReturnType,
   type WaitForTransactionReceiptReturnType,
 } from "viem";
-import { useBlock } from "wagmi";
+import { useBlock, useTransaction } from "wagmi";
 
-import { erc20Abi, l2ToL1MessagePasserAbi, valueEventAbi } from "../abi";
+import {
+  erc20Abi,
+  l2ToL1MessagePasserAbi,
+  optimismPortalAbi,
+  valueEventAbi,
+} from "../abi";
 import { rollupChain, parentChain } from "../config";
-import { titleCase, txType, blockExplorerURL } from "../utils";
+import {
+  titleCase,
+  txType,
+  blockExplorerURL,
+  isCustomGasToken,
+} from "../utils";
 import { WithdrawalActions } from "./WithdrawalActions";
 import { WithdrawalProgressIcon } from "./icons/WithdrawalProgress";
 
 const parseLogs = (transaction: WaitForTransactionReceiptReturnType) => {
   switch (txType(transaction)) {
     case "deposit":
-      return parseEventLogs({
-        abi: erc20Abi,
-        eventName: "Transfer",
-        logs: transaction.logs,
-      });
+      if (isCustomGasToken()) {
+        return parseEventLogs({
+          abi: erc20Abi,
+          eventName: "Transfer",
+          logs: transaction.logs,
+        });
+      } else {
+        return parseEventLogs({
+          abi: optimismPortalAbi,
+          eventName: "TransactionDeposited",
+          logs: transaction.logs,
+        });
```

Comment:
why do we handle custom gas token differently here? both cases generate a TransactionDeposited event from OptimismPortal, right?

Reply:
They both do emit `TransactionDeposited` but the ETH deposit transaction has a value associated with it, where the CGT deposit transaction does not, so if you parse it as `TransactionDeposited`, it doesn't have any extra information about the callData at all. If we parse it as `Transfer` though, you get access to the call data in the logs and can extract the value argument that's passed to the contract to use to display the `value` to the user.